bundles/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/internal/handler/FroniusBaseThingHandler.java

```diff
@@ -160,59 +149,75 @@ protected void updateChannel(String channelId) {
     protected abstract Object getValue(String channelId);
 
     /**
-     * do whatever a thing must do to update the values
+     * Called by the bridge to fetch data and update channels
+     *
+     * @param bridgeConfiguration the connected bridge configuration
+     */
+    public void refresh(FroniusBridgeConfiguration bridgeConfiguration) {
+        try {
+            handleRefresh(bridgeConfiguration);
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
Maybe set ONLINE only if not already ONLINE

Reply:
added the check